# Exercises - Chap. 4

## Ex. 6, p 191

#### (a) Estimate the probability that a student who studies for 40 h and has an undergrad GPA of 3.5 gets an A in the class.

In [3]:
log_r <- (-6+0.05*40+1*3.5)
log_r

[1] -0.5

In [4]:
proba <- exp(log_r)/(1+exp(log_r))
proba

[1] 0.3775407

The probability that a student who studies for 40h and has an undergrad GPA gets and A in the class is 37.75%.

#### (b) How many hours would the student in part (a) need to study to have a 50% chance of getting an A in the class?

In [5]:
f <- function(x) {exp(-6+0.05*x+1*3.5)/(1+exp(-6+0.05*x+1*3.5))}

In [6]:
result <- uniroot(function(x) f(x) - 0.5, c(0,100))

In [7]:
result$root

[1] 50

A student would need to study 50 hours to have a 50% chance of getting an A in the class.

# Ex. 8, p 192

#### Based on these results, which method should we prefer to use for classification of new observations? Why?

Even if the logistic regression method has a lower error rate on the training data set (20%), it has a higher rate on the test set (30%), this indicates that the method is overfitting to the training data. Compared to the 1-nearest neighbors method which has a training and a test error rate of approximatively 18%, the logistic regression's difference between the test error rate and the training error rate shows that it is not generalizing well to new observations. 

However, the 1-nearest neighbors method has a relatively consistent error rate on both data sets, so it could be easily generalize to new observations. So the method we should prefer for classification of new observations is the 1-nearest neighbors method.

# Ex. 9, p 192

#### (a) On average, what fraction of people with an odds of 0.37 of defaulting on their credit card payment will in fact default?

In [13]:
odds <- 0.37
p_odds <- odds/(1+odds)
p_odds

[1] 0.270073

On average, about 27% of people with an odds of 0.37 of defaulting on their credit card payment will actually default.

#### (b) Suppose that an individual has a 16% chance of defaulting on her credit card payment. What are the odds that she will default?

In [14]:
p_odds <- 0.16
odds <- p_odds/(1-p_odds)
odds

[1] 0.1904762

The odds for an individual with 16% chance of defaulting on her credit card payment that she will in fact default is 0.19.